src/App.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import './App.css';
+import { Navigate, Route, Routes } from 'react-router-dom';
+import { useCookies } from 'react-cookie';
+import { projectConfigurationPath, selectedProjectCookieName } from './const';
+import SpinnerLayout from './Components/SpinnerLayout';
+import Metadata from './Components/Metadata';
+import qs from 'qs';
+import Header from './Components/Header';
+import Footer from './Components/Footer';
+import Home from './Pages/Home';
+import Store from './Pages/Store';
+import Articles from './Pages/Articles';
+import Article from './Pages/Article';
+import About from './Pages/About';
+import { useIntl } from 'react-intl';
+import Cafes from './Pages/Cafes';
+import Contact from './Pages/Contacts';
+import Coffee from './Pages/Coffee';
+import Brewer from './Pages/Brewer';
+
+
+interface AppProps{
+  language: string,
+  changeLanguage: (newLanguage: any, newUrl: any) => void
+}
+
+const App: React.FC<AppProps> = (props) => {
+  const [cookie] = useCookies([selectedProjectCookieName])
+  const { formatMessage } = useIntl();
+
+  if (!cookie.ProjectId) {
+    return <Navigate to={projectConfigurationPath} />;
+  }
+
+  const { language, changeLanguage} = props;
+  // slice(1) removes the `?` at the beginning of `location.search`
+  const infoMessage = qs.parse(window.location.search.slice(1)).infoMessage;
+  return (
+    <SpinnerLayout>
+      <div className="application-content">
+        <Metadata />
+        <Header
+          language={language}
+          changeLanguage={changeLanguage}
```

Comment:
Tyoe of changeLanguage can be unified between header and LocalizedApp component.

Reply:
[f412270](https://github.com/Kentico/kontent-sample-app-react/pull/206/commits/f412270ef5b7fdd03c231a6c2cd72aa7dc40bf2f)

Reply:
see Header.tsx comment